docs/engine.md

```diff
@@ -172,6 +174,71 @@ engine.addOperator('startsWithLetter', (factValue, jsonValue) => {
 engine.removeOperator('startsWithLetter');
 ```
 
+### engine.addSharedCondition(String name, Object conditions)
+
+Adds a shared condition to the engine. Rules may include references to this shared condition.
+
+```javascript
+engine.addSharedCondition('validLogin', {
```

Comment:
I was doing some googling to try to see what this was called in other rules engines and there doesn't seem to be a good name for these in wide use. So I'm ok with `condition` as the name I actually like that in a future state this could be used to add new `Condition` subclass constructors which would let make it even more extensible.